bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```diff
@@ -507,6 +507,20 @@ protected State getScaled(Optional<? extends Number> value, Optional<Short> scal
         return getScaled(value.get().longValue(), scaleFactor.get(), unit);
     }
 
+    /**
+     * Returns value multiplied by the 10 on the power of scaleFactory
+     *
+     * @param value the value to alter
+     * @param scaleFactor the scale factor to use (may be negative)
+     * @return the scaled value as a DecimalType
+     */
+    protected State getScaled(Optional<? extends Number> value, Short scaleFactor, Unit<?> unit) {
+        if (!value.isPresent()) {
+            return UnDefType.UNDEF;
+        }
+        return getScaled(value.get().longValue(), scaleFactor, unit);
```

Comment:
It is better to use what you already have. Although you could easily have the other method call this one instead.
```suggestion
        return getScaled(value, Optional.of(scaleFactor), unit);
```

Reply:
Thanks I was aware of that, but that would lead to a different implementation of the getScaled, wich is incorrect in the case of mandatory fields. Mandatory fields will never output an UnDefType value.

Reply:
> Mandatory fields will never output an UnDefType value.

I don't understand, your code clearly returns and UnDefType if it is missing a value...

Reply:
ignore my previous commit, I did misunderstand you. Now I see what you meant: there is a code duplication even if it's an `if`, and you are absolutely right.
I removed the duplication. By the way I removed the unitless getScaled methods, I do not intend to use them anymore.